### PR TITLE
Apm 399 create hello world api

### PR DESCRIPTION
**Pull request by paulwalkington**
## Summary
* Routine Change

Updated Readme file to have more information about where to get environment variable values from. These environment variables are used during to the deploy to apigee process.

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: .github/workflows/continous-integration-workflow.yaml [hunk 1]
```diff
@@ -105,47 +102,48 @@ jobs:
...
-      #   if: github.ref == 'refs/heads/master'
-      #   run: cat newman/*.html > dist/sandbox-tests.html && rm newman/*.html
-
-      # - name: Store sandbox test result artifact (master only)
-      #   if: github.ref == 'refs/heads/master'
-      #   uses: actions/upload-artifact@v1
-      #   with:
-      #     name: sandbox-tests-${{ env.SPEC_VERSION }}+${{ github.run_id }}.html
-      #     path: ./dist/sandbox-tests.html
+      #TODO uncomment when ready to deploy
```
Comment (Sourceless) [inline .github/workflows/continous-integration-workflow.yaml]: Remove

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,62 +1,7 @@
...
-* Added a better example for address lines
-* Removed ods-site-code as a possible value for code system to identify a nominated pharmacy on Patient.
-* Fix some mistakes in the README that referred to a nonexistent directory: `publish` -> `dist`
-
-## 2020-01-30
-* Added automatic version calculation
-* `make publish` now adds version into output oas file
-* Added automatic version tagging to CI pipeline
-* Added changelog
+## 2020-03-06
```
Comment (Sourceless) [inline CHANGELOG.md]: Cacth up date and update changelog

File: README.md [hunk 3]
```diff
@@ -50,11 +50,10 @@ There are `make` commands that alias some of this functionality:
  * `publish` -- Outputs the specification as a **single file** into the `dist/` directory
  * `serve` -- Serves a preview of the specification in human-readable format
  * `generate-examples` -- generate example objects from the specification
- * `validate` -- validate generated examples against FHIR R4
 
 ### Running tests
 #### End-to-end tests
-To run tests, you need to supply an environment. A `local` environment and an environment template are included under `tests/e2e/environments`.
+To run tests, you need to supply an environment. A `local` environment and an environment template is included under `tests/e2e/environments`.
```
Comment (Sourceless) [inline README.md]: `are` was correct because it's talking about two distinct things

File: README.md [hunk 4]
```diff
@@ -116,27 +115,76 @@ Procedure:
...
+* `API_TEST_URL`
+
+### github deployment
+
+github uses github actions to deploy the code to apigee. The github action uses secrets to populate environment variables.
+You need a github secret for each environment variable. Each of the above environment variables need an equivalent secret in github for
+the deployment to work. These are pre-populated for you [here](https://github.com/NHSDigital/hello-world-api/settings/secrets/new). 
+If you get a 404 for this page you will need to update your github account permissions.
+
+### local deployment
```
Comment (Sourceless) [inline README.md]: Titlecase please

File: apiproxy/hello-world-v1.xml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<APIProxy name="pds">
+<APIProxy name="hello-world-v1">
```
Comment (Sourceless) [inline apiproxy/hello-world-v1.xml]: just `hello-world` is fine

File: apiproxy/policies/AssignMessage-AddCorsHeaders.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<AssignMessage async="false" continueOnError="false" enabled="true" name="AssignMessage-AddCorsHeaders">
+    <Set>
+        <Headers>
+            <Header name="Access-Control-Allow-Origin">{request.header.origin}</Header>
+            <Header name="Access-Control-Allow-Headers">origin, x-requested-with, accept, content-type, authorization, apikey</Header>
+            <Header name="Access-Control-Max-Age">3628800</Header>
```
Comment (Sourceless) [inline apiproxy/policies/AssignMessage-AddCorsHeaders.xml]: CORS Headers should not be valid for more than 5-10 mins

File: pyproject.toml [hunk 7]
```diff
@@ -5,22 +5,20 @@ python = "^3.8"
 [tool.poetry]
 name = "apim-definitions"
```
Comment (Sourceless) [inline pyproject.toml]: Please set the name here

File: pyproject.toml [hunk 8]
```diff
@@ -5,22 +5,20 @@ python = "^3.8"
...
-description = "FHIR PDS API Definitions"
+description = "Hello World API Definitions"
 
 license = "MIT"
 
 authors = [
-    "Laurence Pakenham-Smith <laurence.smith2@nhs.net>",
-    "Alex Hawdon <alex.hawdon1@nhs.net>",
-    "Tony Heap <tony.heap1@nhs.net>"
+    #TODO add authors
```
Comment (Sourceless) [inline pyproject.toml]: You can add yourselves :) just keep the original list as well

File: sandbox/package.json [hunk 9]
```diff
@@ -1,23 +1,16 @@
...
-        "@hapi/joi": "^17.1.0",
-        "fast-json-patch": "^3.0.0-1",
-        "inert": "^5.1.3",
-        "lodash": "^4.17.15",
-        "nhs-number-validator": "^1.1.1"
-    },
-    "devDependencies": {
-        "eslint": "^6.8.0"
-    }
+  "name": "mock",
```
Comment (Sourceless) [inline sandbox/package.json]: change to something like `hello-world-api-sandbox`

File: scripts/deploy_proxy.sh [hunk 10]
```diff
@@ -7,8 +7,10 @@ set -o nounset errexit pipefail
...
 rm -rf dist/apiproxy
-cp -R apiproxy/ dist/
-sed -i "s/PROXY_BASE_PATH/$APIGEE_BASE_PATH/g" dist/apiproxy/proxies/default.xml
+cp -Rv apiproxy dist/
+sed "s/PROXY_BASE_PATH/$APIGEE_BASE_PATH/g" dist/apiproxy/proxies/default.xml > dist/apiproxy/proxies/default_temp.xml
+cp -f dist/apiproxy/proxies/default_temp.xml dist/apiproxy/proxies/default.xml
+rm -rf dist/apiproxy/proxies/default_temp.xml
 mkdir -p dist/apiproxy/resources/hosted
-rsync -av --copy-links --exclude="node_modules" sandbox/ dist/apiproxy/resources/hosted
+rsync -av --copy-links --exclude="node_modules" --filter=':- .gitignore' sandbox/ dist/apiproxy/resources/hosted
```
Comment (Sourceless) [inline scripts/deploy_proxy.sh]: We should backport this to pds api :)

File: specification/hello-world-api.yaml [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    Use this API alongside our [tutorials](https://digital.nhs.uk/developer/api-tutorials) to learn how to connect to our APIs.
+
+    ## Endpoints
+    To see details for specific endpoints, select from the list on the left hand side of the page:
+    - `GET /hello/world` - an example of an open-access endpoint
+    - `GET /hello/application` - an example of an application-restricted endpoint
+    - `GET /hello/user` - an example of a user-restricted endpoint
+  contact:
+    name: TBC
+    email: tbc@nhs.net
```
Comment (Sourceless) [inline specification/hello-world-api.yaml]: `api.management@nhs.net`

File: specification/hello-world-api.yaml [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+    ## Overview
+    Use this API alongside our [tutorials](https://digital.nhs.uk/developer/api-tutorials) to learn how to connect to our APIs.
+
+    ## Endpoints
+    To see details for specific endpoints, select from the list on the left hand side of the page:
+    - `GET /hello/world` - an example of an open-access endpoint
+    - `GET /hello/application` - an example of an application-restricted endpoint
+    - `GET /hello/user` - an example of a user-restricted endpoint
+  contact:
+    name: TBC
```
Comment (Sourceless) [inline specification/hello-world-api.yaml]: API Management Support

File: CHANGELOG.md (hunk 2 above)
Comment (paulwalkington) [reply CHANGELOG.md]: done

**Pull request merged**